students/ltrisolini/Lesson09/mailroom_OO.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python3
+import sys
+from collections import defaultdict
+
+class Donor:
+
+    def __init__(self, name, donations=None):
+        self.name = name
+        if donations is None:
+            self.donations = []
+        else:
+            self.donations = donations
+
+
+    def add_donation(self, amount):
+        try:
+            self.donations.append(int(amount))
+        except ValueError:
+            print("Please enter a number!")
+
+
+    def number_donations(self):
+        return len(self.donations)
+
+
+    def total_donation(self):
+        try:
+            return sum(self.donations)
+        except TypeError:
+            return self.donations
+
+    def avg_donation(self):
+        try:
+            return self.total_donation()/ self.number_donations()
+        except TypeError:
+            return self.donations
+
+class Roster:
+
+    def __init__(self, donors=None):
+        self.donor_list = donors
```

Comment:
since you allow `donors` to be None, it looks like the code can break down the line when looping over `self.donor_list`